### PR TITLE
Add support to update account

### PR DESCRIPTION
**Pull request by remilapeyre**
Close https://github.com/go-acme/lego/issues/443

### CODE REVIEW

**Comment by remilapeyre** [comment]
Hi @Idez, I see that some tests failed in the CI but I'm not sure why they failed. Are those flaky tests or did I break something?

File: e2e/challenges_test.go [hunk 1]
```diff
@@ -198,6 +198,37 @@ func TestChallengeTLS_Run_Revoke_Non_ASCII(t *testing.T) {
 	}
 }
 
+func TestRegistrar_UpdateAccount(t *testing.T) {
```
Comment (cpu) [inline e2e/challenges_test.go]: One drive-by suggestion: This unit test might also benefit from a test case that verifies **removing** all contact addresses with an `UpdateRegistration` operation works as expected.

**Comment by ldez** [comment]
@remilapeyre to fix the CI, you need to rebase your PR.

File: e2e/challenges_test.go (hunk 1 above)
Comment (remilapeyre) [reply e2e/challenges_test.go]: Removing contact addresses is not supported at the moment, I wish to add it later but it will change the User interface.

**Comment by remilapeyre** [comment]
Oops, I will do this tomorrow

**Comment by remilapeyre** [comment]
Hi @Idez, it should be good now
